### DISCUSSION TITLE
MPI on a single PC with multiple cores

### DISCUSSION THREAD

**Discussion by Srinivas-Ganti** [NanoComp/meep#2252] [Q&A] [answered]
Hello, I am experimenting with ways to speed up my 3D simulations in meep. Is this possible to do using MPI on a single machine with multiple cores? I am using a modest computing node with Intel i10 (8 cores), 64 GB RAM, and running Ubuntu. However, I don't think I am utilizing the full available computing power when I run my simulation script from my IDE or terminal. I noticed the usage of mpirun on the [simpetus projects page](http://www.simpetus.com/projects.html#meep_image_sensor)

Does this same usage apply to my single node case too, e.g.

` mpirun -np 8 python3 myHugeSim.py`

Will this add any computational speed benefit or am I just running a copy of the program in each of the jobs from the 8 requested cores? Is there a way to properly parallelize the job to increase computational efficiency on my machine?

**Comment by smartalecH** [comment, marked as answer]
> mpirun -np 8 python3 myHugeSim.py



Yes that should work. 

See the docs: https://meep.readthedocs.io/en/latest/Parallel_Meep/#using-parallel-meep
